### DISCUSSION TITLE
a8r8 on windows

### DISCUSSION THREAD

**Discussion by aolko** [ramyma/a8r8#23] [General]
Wow, what a SD UI we have here.

First of all, this code is written in an obscure language, Elixir.
Secondly, it may be helpful to check if the __core dependencies__ are compatible with Windows, as it is the most widely used operating system (https://github.com/akash-akya/vix/issues/79). It seems that the code may not build successfully because it is not compatible with the nmake in VC++ build tools.
Lastly, relying on WSL for this may be a limitation, especially if it cannot start under NAT or in mirrored mode (https://github.com/ramyma/a8r8/issues/22). As for the windows firewall - it's completely disabled.

**Comment by ramyma** [comment]
Hello,
Well, like you mentioned, there's a dependency that's not compatible with Windows, and replacing it is not straight forward as it's used for processing the input and output images in different places in the code.
I was planning to look for alternatives, but as you may know research needs time.

As for the WSL situation, different users had varying results; some went through the steps with ease, and some encountered some issues that required some extra steps to get things wired correctly. 

On that note, I appreciate your feedback as this is the only way for me to make the installation steps better.

With NAT mode, the run script figures out your WSL virtual route automatically, but in some cases you need to modify some firewall settings (yes, even when it's completely turned off!)

Here're some steps that resolved some issues that some users were facing:

1. From start menu, open Windows Defender Firewall with Advanced Security
2. Click on Windows Defender Firewall root item on the left
3. Click on Windows Defender Firewall Properties
4. Choose the Private Profile tab
5. Under the firewall state box, click Customize
6. Uncheck vEthernet (WSL) and press OK
7. Do the same steps with the Public Profile tab
8. Press OK to complete it

I haven't tried mirrored mode, but conceptually you can set addresses to resolve each backend in user.sh; just uncomment the corresponding line in `user.sh` and restart the application.

It also helps if you can mention your windows version, build number and WSL version.

Finally, I'm not sure what you meant by Elixir being an obscure language, it's not as widely used as Python if that's what you meant.

Cheers!

**Reply by aolko** [reply]
> It also helps if you can mention your windows version, build number and WSL version.

WSL: 2.2.4.0
Core: 5.15.153.1-2
WSLg: 1.0.61
MSRDC: 1.2.5326
Direct3D: 1.611.1-81528511
DXCore: 10.0.26091.1-240325-1447.ge-release
Windows: 10.0.22621.3737 (11)

> Finally, I'm not sure what you meant by Elixir being an obscure language, it's not as widely used as Python if that's what you meant.

you'd have much more luck with go or rust or c or python or some other mainstream language and it's stdlib/packages

**Comment by ramyma** [comment]
Did you get to try the steps that I provided?

**Reply by aolko** [reply]
no dice so far, however, i didn't reboot yet

**Reply by aolko** [reply]
doesn't work

**Reply by ramyma** [reply]
Can you provide further info on what you tried and how it's not working?

**Reply by aolko** [reply]
see above,
the host is MIA outside WSL

**Reply by ramyma** [reply]
which steps did you try?

**Reply by aolko** [reply]
all of them

**Reply by ramyma** [reply]
We can go through it on Discord, to be able to better understand what it's stuck on.
https://discord.gg/RzdGa839MK
